enterprise/backend/src/metabase_enterprise/workspaces/common.clj

```diff
@@ -3,121 +3,125 @@
    [metabase-enterprise.workspaces.isolation :as ws.isolation]
    [metabase-enterprise.workspaces.mirroring :as ws.mirroring]
    [metabase.api-keys.core :as api-key]
-   [metabase.util.log :as log]
-   [metabase.util.malli :as mu]
-   [metabase.util.secret :as u.secret]
+   [metabase.api.common :as api]
    [toucan2.core :as t2]))
 
-;; TODO: Delete when we have start using dag.
-#_(defn- make-graph-from-stuff [_stuff]
-    {})
+(defn- temp-assert-single-transform!
+  "A temporary restriction, until we have graph analysis merged."
+  [upstream-entities]
+  (assert (and (= 1 (count (keys upstream-entities)))
+               (= 1 (count (:transforms upstream-entities))))
+          "The only contents of the workspace must be a single transform."))
 
-#_{:clj-kondo/ignore [:unused-private-var]}
-(defn- dummy-dag-add-mirrored-entities
-  [x]
-  x)
-
-(defn- dummy-dag-single-transform->entities-info
-  [stuffs]
-  (let [transform-id (first (get stuffs :transforms))
-        transform (t2/select-one :model/Transform :id transform-id)
-        _ (assert (not-empty transform))
-        table-id (get-in (t2/select-one :model/Transform :id 1 #_transform-id)
-                         [:source :query :stages 0 :source-table])
-        _ (assert (pos-int? table-id))
-        table (t2/select-one :model/Table :id table-id)
-        _ (assert (not-empty table))]
-    {:transforms [transform]
-     :inputs [table]}))
+(defn- build-graph
+  "Placeholder for our actual dag module, handling at most 1 transform, and not returning all the analysis."
+  [upstream]
+  (if (not-any? seq (vals upstream))
+    {:db_id      nil
+     :transforms []
+     :inputs     []}
+    (do
+      (temp-assert-single-transform! upstream)
+      (let [transform-id (first (get upstream :transforms))
+            ;; NOte: we only support MBQL transforms for now
+            transform    (t2/select-one [:model/Transform :id :name :source :target] :id transform-id)
+            _            (assert (not-empty transform))
+            ;; Note: if there are additional tables being joined, we are not picking them up @_@
+            source-id    (get-in transform [:source :query :stages 0 :source-table])
+            _            (assert (pos-int? source-id))
+            source-table (t2/select-one [:model/Table :id :schema :name] :id source-id)
+            _            (assert (not-empty source-table))
+            output-table (let [{:keys [schema name]} (:target transform)]
+                           ;; TODO (Chris 2025-11-19) Relax downstream assumption that upstream-tx output table exists
+                           {:id     (t2/select-one-pk :model/Table :schema schema :name name)
+                            :schema schema
```

Comment:
We could include the db id here in all of them, even though it's redundant.

It would allow us to remove the "everything is in the same dwh" assumption in a few places, and also might make things easier for Clickhouse. I decided not to do this yet.